### DISCUSSION TITLE
[Q] Webserial (psram) with second esp32-debug device

### DISCUSSION THREAD

**Discussion by DaeMonSxy** [mathieucarbou/MycilaWebSerial#16] [General]
@mathieucarbou  
What do you think, would it be possible to create code (extend yours?) to connect a second esp32 to get serial outputs from main esp32/esp8266 device.
Kind of **esp32-debug device** replacing  serial connect pc, or raspi?
Upon connection to esp32-debug device it can show x lines of past serial messages, then on-time show the current ongoing serial communication?

Even, maybe the log of received responses in psram can be downloaded as log file?

(thinking of having a such a device for debuging a while now, replacing raspi-s running now on linux and screen for logging)

**Comment by mathieucarbou** [comment]
Hello,

Through the Network remotely or by connecting an esp on Rx/tx ?

This seems totally another project though (sort of bridge API).

This project is meant to stream log lines onto websocket through the use of the print API.

What you would do is a print implementation that will be used on the emitter and on the receptor side, webseeial (Print API) could be passed to the lib so that the lib forwards the readings to it.

**Reply by DaeMonSxy** [reply]
good queries, that is the reason I rised the question :).

Well, usb to usb ?

My automation esp32 controllers has usb ports ( or serial converters, not metter yet), 
Can we use usb ports in both esps, let say usbc-usbc one is sending Serial.prints , another is Serial.reading until end of line? through usbc cable.

**I would use few blocks**

1 connectiong to remote esp32 on usbc
2 reading remote-esp32 serial prints
3 saving reads to psram
4 saving to whatever, daily logs....
5 if connected send back x lines of log (your lib ?)
6 if connected send back current serial.readings (your lib for sure)
7 downloading logs (kind of file listing)

What do you think?

**Reply by mathieucarbou** [reply]
I think that this is already doable:

- on the emitter: write logs to serial as usual (you can even use my MycilaLogger library if you need and see how I use it in my projects to forward the logs either to it or to the esp logging system, 
- on the esp32 that is plugged, the uart linked to the emitter will receive the sent data, you just need to read it line per line and forward to webserial.

This is possible also to use the Stream debugger library (see my GitHub profile) to intercept the stream if you need.

**Reply by mathieucarbou** [reply]
But there is no need for any special Integration inside this lib to achieve that I think.
